holidays/countries/laos.py

```diff
@@ -0,0 +1,447 @@
+#  python-holidays
+#  ---------------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import JAN, MAR, APR, MAY, JUL, OCT, DEC
+from holidays.calendars.thai import KHMER_CALENDAR
+from holidays.constants import BANK, PUBLIC, SCHOOL, WORKDAY
+from holidays.groups import InternationalHolidays, ThaiCalendarHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    THU_FRI_SAT_TO_NEXT_MON_TUE,
+    FRI_SAT_SUN_TO_NEXT_TUE_WED,
+    SAT_SUN_TO_NEXT_MON,
+)
+
+
+class Laos(ObservedHolidayBase, InternationalHolidays, ThaiCalendarHolidays):
+    """
+    A subclass of :py:class:`HolidayBase` representing public holidays in Laos.
+
+    References:
+
+    - Based on: https://en.wikipedia.org/wiki/Public_holidays_in_Laos
+                https://www.ayutthaya.go.th/5786-%20ข้อควรรู้และวันหยุดในประเทศลาว
+                ດຳລັດວ່າດ້ວຍ ວັນພັກ ເລກທີ 386 /ລບ (15.12.2017)
```

Comment:
We can probably eliminate the Thai source here; I used it prior to discovering other referencing sources  (LSX, BCEL) - [`ດຳລັດວ່າດ້ວຍ ວັນພັກ ເລກທີ 386 /ລບ (15.12.2017)`](https://juristact.weebly.com/uploads/1/0/9/9/109947087/d17_386.pdf) though is the holiday law code itself with no readily-translated English language sources anywhere.

Reply:
Would something like "Decree on Holidays No. 386 / Rev. 15.12.2017" work here?. Not sure if that's the correct translation. I just want to make it useful for most of the code readers. 

Just in case -- it's fine to have it in URL as you can't do anything about that :)

Reply:
Let's go with that then 👍